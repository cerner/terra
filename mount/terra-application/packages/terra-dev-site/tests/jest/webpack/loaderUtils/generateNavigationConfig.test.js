const path = require('path');
const generatePagesConfig = require('../../../../src/webpack/loaderUtils/generateNavigationConfig');

describe('generateNavigationConfig', () => {
  it('test standard config + additional content', () => {
    const addContextDependency = jest.fn();
    const config = generatePagesConfig({
      siteConfig: {
        additionalSearchDirectories: [],
        primaryNavigationItems: [{
          path: '/test',
          label: 'Test',
          contentExtension: 'ext',
          additionalContent: [
            {
              label: 'add',
              filePath: path.resolve(__dirname, 'testContent', 'additionalContent.md'),
            },
          ],
        }, {
          path: '/findNothing',
          label: 'findNothing',
          contentExtension: 'findNothing',
        }],
        sourceFolder: 'tests/jest/webpack/loaderUtils/testContent',
        distributionFolder: '',
        enableDebugLogging: false,
        namespace: 'npm-package',
      },
      resolveExtensions: ['.js'],
      mode: 'development',
      contentDirectory: 'TestA',
      isLernaMonoRepo: true,
      addContextDependency,
      logger: () => {},
    });
    expect(addContextDependency).toHaveBeenCalledWith(`${path.resolve(__dirname, 'testContent', 'TestA')}/`);
    expect(config.navigationConfig).toMatchSnapshot();
    expect(config.routesMap).toMatchSnapshot();
    expect(config.pageConfig).toMatchSnapshot();
    expect(config.contentImports).toEqual({
      '/test/cerner-terra-dev-site/file': path.join(__dirname, '/testContent/TestA/ext/file.ext.js'),
      '/test/cerner-terra-dev-site/file-outside': path.join(__dirname, 'testContent/TestA/fileOutside.ext.md'),
      '/test/cerner-terra-dev-site/add': path.join(__dirname, 'testContent/additionalContent.md'),
    });
  });

  it('tests prod mode', () => {
    const addContextDependency = jest.fn();
    const config = generatePagesConfig({
      siteConfig: {
        additionalSearchDirectories: [],
        primaryNavigationItems: [{
          path: '/test',
          label: 'Test',
          contentExtension: 'ext',
        }],
        distributionFolder: 'tests/jest/webpack/loaderUtils/testContent',
        sourceFolder: '',
        enableDebugLogging: false,
        namespace: 'npm-package',
      },
      resolveExtensions: ['.js'],
      mode: 'production',
      contentDirectory: 'TestA',
      isLernaMonoRepo: true,
      addContextDependency,
      logger: () => {},
    });
    expect(addContextDependency).not.toHaveBeenCalled();
    expect(config.navigationConfig).toMatchSnapshot();
    expect(config.routesMap).toMatchSnapshot();
    expect(config.pageConfig).toMatchSnapshot();
    expect(config.contentImports).toEqual({
      '/test/cerner-terra-dev-site/file': path.join(__dirname, 'testContent/TestA/ext/file.ext.js'),
      '/test/cerner-terra-dev-site/file-outside': path.join(__dirname, 'testContent/TestA/fileOutside.ext.md'),
    });
  });

  it('tests sorting', () => {
    const addContextDependency = jest.fn();
    const config = generatePagesConfig({
      siteConfig: {
        additionalSearchDirectories: [],
        primaryNavigationItems: [{
          path: '/test',
          label: 'Test',
          contentExtension: 'ext',
        }],
        distributionFolder: '',
        sourceFolder: 'tests/jest/webpack/loaderUtils/testContent',
        enableDebugLogging: false,
        namespace: 'npm-package',
      },
      resolveExtensions: ['.js'],
      mode: 'development',
      contentDirectory: 'TestB',
      isLernaMonoRepo: true,
      addContextDependency,
      logger: () => {},
    });
    expect(addContextDependency).toHaveBeenCalledWith(`${path.resolve(__dirname, 'testContent', 'TestB')}/`);
    expect(config.navigationConfig).toMatchSnapshot();
    expect(config.routesMap).toMatchSnapshot();
    expect(config.pageConfig).toMatchSnapshot();
    expect(config.contentImports).toEqual({
      '/test/cerner-terra-dev-site/a': path.join(__dirname, 'testContent/TestB/ext/a.ext.js'),
      '/test/cerner-terra-dev-site/b/file': path.join(__dirname, 'testContent/TestB/ext/b/file.ext.js'),
      '/test/cerner-terra-dev-site/dir/file': path.join(__dirname, 'testContent/TestB/ext/dir.c/file.ext.js'),
      '/test/cerner-terra-dev-site/first-file': path.join(__dirname, 'testContent/TestB/ext/firstFile.a.ext.js'),
      '/test/cerner-terra-dev-site/second-file': path.join(__dirname, 'testContent/TestB/ext/secondFile.b.ext.js'),
    });
  });

  it('tests additionalSearchDirectories', () => {
    const addContextDependency = jest.fn();
    const config = generatePagesConfig({
      siteConfig: {
        additionalSearchDirectories: [
          path.resolve(__dirname, 'testContent', 'TestA'),
        ],
        primaryNavigationItems: [{
          path: '/test',
          label: 'Test',
          contentExtension: 'ext',
        }],
        distributionFolder: '',
        sourceFolder: 'tests/jest/webpack/loaderUtils/testContent',
        enableDebugLogging: false,
        namespace: 'npm-package',
      },
      resolveExtensions: ['.js'],
      mode: 'development',
      contentDirectory: 'Empty',
      isLernaMonoRepo: true,
      addContextDependency,
      logger: () => {},
    });
    expect(config.navigationConfig).toMatchSnapshot();
    expect(config.routesMap).toMatchSnapshot();
    expect(config.pageConfig).toMatchSnapshot();
    expect(config.contentImports).toEqual({
      '/test/cerner-terra-dev-site/file': path.join(__dirname, 'testContent/TestA/ext/file.ext.js'),
      '/test/cerner-terra-dev-site/file-outside': path.join(__dirname, 'testContent/TestA/fileOutside.ext.md'),
    });
  });

  it('tests multiple primary nav items', () => {
    const addContextDependency = jest.fn();
    const config = generatePagesConfig({
      siteConfig: {
        additionalSearchDirectories: [],
        primaryNavigationItems: [{
          path: '/test',
          label: 'Test',
          contentExtension: 'ext',
        }, {
          path: '/doc',
          label: 'Doc',
          contentExtension: 'doc',
        }],
        sourceFolder: 'tests/jest/webpack/loaderUtils/testContent',
        distributionFolder: '',
        enableDebugLogging: false,
        namespace: 'npm-package',
      },
      resolveExtensions: ['.js'],
      mode: 'development',
      contentDirectory: 'TestC',
      isLernaMonoRepo: true,
      addContextDependency,
      logger: () => {},
    });
    expect(addContextDependency).toHaveBeenCalledWith(`${path.resolve(__dirname, 'testContent', 'TestC')}/`);
    expect(config.navigationConfig).toMatchSnapshot();
    expect(config.routesMap).toMatchSnapshot();
    expect(config.pageConfig).toMatchSnapshot();
    expect(config.contentImports).toEqual({
      '/test/cerner-terra-dev-site/file': path.join(__dirname, 'testContent/TestC/ext/file.ext.js'),
      '/doc/cerner-terra-dev-site/file': path.join(__dirname, 'testContent/TestC/doc/file.doc.js'),
    });
  });
});
