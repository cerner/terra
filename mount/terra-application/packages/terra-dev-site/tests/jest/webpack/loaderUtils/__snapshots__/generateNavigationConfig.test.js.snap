// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateNavigationConfig test standard config + additional content 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "label": "Add",
        "path": "/test/cerner-terra-dev-site/add",
        "type": "md",
      },
      Object {
        "label": "File",
        "path": "/test/cerner-terra-dev-site/file",
        "type": "js",
      },
      Object {
        "label": "File Outside",
        "path": "/test/cerner-terra-dev-site/file-outside",
        "type": "md",
      },
    ],
    "label": "Test",
    "path": "/test",
  },
]
`;

exports[`generateNavigationConfig test standard config + additional content 2`] = `
Object {
  "/": "/test/cerner-terra-dev-site/add",
  "/test": "/test/cerner-terra-dev-site/add",
}
`;

exports[`generateNavigationConfig test standard config + additional content 3`] = `
Object {
  "/test/cerner-terra-dev-site/add": Object {
    "label": "Add",
    "type": "md",
  },
  "/test/cerner-terra-dev-site/file": Object {
    "label": "File",
    "type": "js",
  },
  "/test/cerner-terra-dev-site/file-outside": Object {
    "label": "File Outside",
    "type": "md",
  },
}
`;

exports[`generateNavigationConfig tests additionalSearchDirectories 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "label": "File",
        "path": "/test/cerner-terra-dev-site/file",
        "type": "js",
      },
      Object {
        "label": "File Outside",
        "path": "/test/cerner-terra-dev-site/file-outside",
        "type": "md",
      },
    ],
    "label": "Test",
    "path": "/test",
  },
]
`;

exports[`generateNavigationConfig tests additionalSearchDirectories 2`] = `
Object {
  "/": "/test/cerner-terra-dev-site/file",
  "/test": "/test/cerner-terra-dev-site/file",
}
`;

exports[`generateNavigationConfig tests additionalSearchDirectories 3`] = `
Object {
  "/test/cerner-terra-dev-site/file": Object {
    "label": "File",
    "type": "js",
  },
  "/test/cerner-terra-dev-site/file-outside": Object {
    "label": "File Outside",
    "type": "md",
  },
}
`;

exports[`generateNavigationConfig tests multiple primary nav items 1`] = `
Array [
  Object {
    "label": "Test",
    "pageConfig": Object {
      "label": "File",
      "path": "/test/cerner-terra-dev-site/file",
      "type": "js",
    },
    "path": "/test",
  },
  Object {
    "label": "Doc",
    "pageConfig": Object {
      "label": "File",
      "path": "/doc/cerner-terra-dev-site/file",
      "type": "js",
    },
    "path": "/doc",
  },
]
`;

exports[`generateNavigationConfig tests multiple primary nav items 2`] = `
Object {
  "/": "/test/cerner-terra-dev-site/file",
  "/doc": "/doc/cerner-terra-dev-site/file",
  "/test": "/test/cerner-terra-dev-site/file",
}
`;

exports[`generateNavigationConfig tests multiple primary nav items 3`] = `
Object {
  "/doc/cerner-terra-dev-site/file": Object {
    "label": "File",
    "type": "js",
  },
  "/test/cerner-terra-dev-site/file": Object {
    "label": "File",
    "type": "js",
  },
}
`;

exports[`generateNavigationConfig tests prod mode 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "label": "File",
        "path": "/test/cerner-terra-dev-site/file",
        "type": "js",
      },
      Object {
        "label": "File Outside",
        "path": "/test/cerner-terra-dev-site/file-outside",
        "type": "md",
      },
    ],
    "label": "Test",
    "path": "/test",
  },
]
`;

exports[`generateNavigationConfig tests prod mode 2`] = `
Object {
  "/": "/test/cerner-terra-dev-site/file",
  "/test": "/test/cerner-terra-dev-site/file",
}
`;

exports[`generateNavigationConfig tests prod mode 3`] = `
Object {
  "/test/cerner-terra-dev-site/file": Object {
    "label": "File",
    "type": "js",
  },
  "/test/cerner-terra-dev-site/file-outside": Object {
    "label": "File Outside",
    "type": "md",
  },
}
`;

exports[`generateNavigationConfig tests sorting 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "label": "First File",
        "path": "/test/cerner-terra-dev-site/first-file",
        "type": "js",
      },
      Object {
        "label": "Second File",
        "path": "/test/cerner-terra-dev-site/second-file",
        "type": "js",
      },
      Object {
        "children": Array [
          Object {
            "label": "File",
            "path": "/test/cerner-terra-dev-site/dir/file",
            "type": "js",
          },
        ],
        "label": "Dir",
      },
      Object {
        "label": "A",
        "path": "/test/cerner-terra-dev-site/a",
        "type": "js",
      },
      Object {
        "children": Array [
          Object {
            "label": "File",
            "path": "/test/cerner-terra-dev-site/b/file",
            "type": "js",
          },
        ],
        "label": "B",
      },
    ],
    "label": "Test",
    "path": "/test",
  },
]
`;

exports[`generateNavigationConfig tests sorting 2`] = `
Object {
  "/": "/test/cerner-terra-dev-site/first-file",
  "/test": "/test/cerner-terra-dev-site/first-file",
}
`;

exports[`generateNavigationConfig tests sorting 3`] = `
Object {
  "/test/cerner-terra-dev-site/a": Object {
    "label": "A",
    "type": "js",
  },
  "/test/cerner-terra-dev-site/b/file": Object {
    "label": "File",
    "type": "js",
  },
  "/test/cerner-terra-dev-site/dir/file": Object {
    "label": "File",
    "type": "js",
  },
  "/test/cerner-terra-dev-site/first-file": Object {
    "label": "First File",
    "type": "js",
  },
  "/test/cerner-terra-dev-site/second-file": Object {
    "label": "Second File",
    "type": "js",
  },
}
`;
